### PR TITLE
Update hardening guide

### PR DESCRIPTION
**Pull request by ozzyjohnson**
Updates the hardening guide to...

- Provide Vagrantfiles for launching in VirtualBox or AWS easily via Vagrant with a secondary disk.
- Unify the instructions post-Vagrantfile.
- Use sgdisk for a simplified, non-interactive partitioning experience.
- Make use of homebrew and cask.
- Fix assorted typos.

Still lacks changes to get away from 'physical' partitioning and backup / copy could be more efficient. Now that the guide and script are in sync, updates of that sort should be easier.

### CODE REVIEW

File: hardening.md [hunk 1]
```diff
@@ -1,137 +1,227 @@
...
-This hardening guide, and the operating system (OS) configuration it produces, is in *early beta* and will change. The baseline is currently version 0.1.2. 18F is currently testing this baseline for deployment into production.
+This hardening guide, and the operating system (OS) configuration it produces, is in *early beta* and will change. The baseline is currently version 0.1.4. 18F is currently testing this baseline for deployment into production.
 
 There are many recommended controls that are already in place with given a fresh install of either Ubuntu 12.04 LTS or Ubuntu 14.04 LTS. These controls are fully itemized at TBD and are covered by our an open source compliance testing suite we are currently working on.
 
-Controls were implemented on both an Ubuntu 12.04 box downloaded from the [VagrantCloud](https://vagrantcloud.com/hashicorp/precise64/version/2/provider/virtualbox.box) and the default 640-bit Ubuntu 14.04 Amazon Machine Image (AMI).
+Controls were implemented on both an Ubuntu 12.04 box downloaded from the [VagrantCloud](https://vagrantcloud.com/hashicorp/precise64/version/2/provider/virtualbox.box) and the default 64-bit Ubuntu 14.04 Amazon Machine Image (AMI).
```
Comment (konklone) [inline hardening.md]: Seems like we're leaving a lot of bits on the table here.

Comment (ozzyjohnson) [inline hardening.md]: As the general thrust here is security, I'm not terribly concerned unless they are sticky.

File: hardening.md [hunk 2]
```diff
@@ -139,81 +229,84 @@ The order of abstraction is PV > VG > LV.
...
+	tmpfs                            52M  373k   52M   1% /run
+	none                            5.3M     0  5.3M   0% /run/lock
+	none                            257M     0  257M   0% /run/shm
+	none                            105M     0  105M   0% /run/user
+	vagrant                         500G   45G  455G   9% /vagrant
+	/dev/mapper/securefolders-temp   11G   24M  9.9G   1% /tmp
+
+### Securing the Filesystem:
 
 Let's put some security options on that folder.
```
Comment (janearc) [inline hardening.md]: s/folder/mountpoint/ ?

**Comment by dlapiduz** [comment]
:+1:

**Comment by dlapiduz** [comment]
:shipit:

**Pull request merged**